src/main/java/gov/nih/nci/evs/api/service/AbstractStardogLoadServiceImpl.java

```diff
@@ -485,6 +499,160 @@ public void loadObjects(
+        // index subsetConcept
+        operationsService.index(subsetConcept, terminology.getIndexName(), Concept.class);
+      }
+      // explicitly set leaf since it defaults to false
+      newSubsetEntry.setLeaf(!newSubsets.containsValue(newSubsetEntry.getCode()));
+      // add extra relevant properties to new subset
+      newSubsetEntry.getProperties().add(new Property("Publish_Value_Set", "Yes"));
+      newSubsetEntry.getProperties().add(new Property("EVSRESTAPI_Subset_Format", "NCI"));
+      // index newSubsetEntry
+      operationsService.index(newSubsetEntry, terminology.getIndexName(), Concept.class);
+      // create new subset for parentSubset to add as child of existing subset
+      Concept parentSubsetChild = new Concept(newSubsetEntry);
+      // strip out everything the subset tree doesn't need
+      ConceptUtils.applyInclude(parentSubsetChild, new IncludeParam("minimal,properties"));
+
+      parentSubset.getChildren().add(parentSubsetChild);
+      if (parentSubset.getProperties().stream()
+          .noneMatch(property -> property.getType().equals("Publish_Value_Set"))) {
+        parentSubset.getProperties().add(new Property("Publish_Value_Set", "Yes"));
+      }
+      parentSubset.getProperties().add(new Property("EVSRESTAPI_Subset_Format", "NCI"));
+      // index parentSubset
+      operationsService.index(parentSubset, terminology.getObjectIndexName(), Concept.class);
+    }
+    logger.info("Extra subsets added");
+  }
+
+  public static List<Sheet> loadExcelSheets(String url, String filepath) {
+    List<Sheet> sheets = new ArrayList<>();
+    Workbook workbook = null;
+
+    try {
+      // Try downloading the file from the provided URL
+      InputStream inputStream = new URL(url).openStream();
+      workbook = new HSSFWorkbook(inputStream);
+      System.out.println("Excel file successfully loaded from URL.");
+    } catch (Exception e) {
+      System.err.println("Failed to download Excel file from URL. Error: " + e.getMessage());
+      try {
+        // If download fails, fall back to the local backup file
+        FileInputStream fileInputStream = new FileInputStream(filepath);
+        workbook = new HSSFWorkbook(fileInputStream);
+        System.out.println("Excel file successfully loaded from backup file.");
+      } catch (Exception fileException) {
+        System.err.println(
+            "Failed to load Excel file from backup file. Error: " + fileException.getMessage());
+        return sheets; // Return an empty list if all attempts fail
+      }
+    }
+
+    // Extract sheets from the workbook
+    if (workbook != null) {
+      for (int i = 0; i < workbook.getNumberOfSheets(); i++) {
+        sheets.add(workbook.getSheetAt(i));
+      }
+      try {
+        workbook.close(); // Close the workbook to release resources
+      } catch (IOException e) {
+        System.err.println("Failed to close the workbook. Error: " + e.getMessage());
+      }
```

Comment:
Never use this pattern of having a try/catch and then just logging the error.  In this case, just define the methdo as "throws Exception" and let it just get thrown.  If there's really a problem here, we should stop.  This is probably true of the try on 629 also.  Why bother even catching it - just let that exception get thrown.  The other option is to catch the exception, then do a logger.error("your own error message",e); and then throw new Exception("Unexpected xxx",e)

Reply:
Catching it I think is going to be important for the audit log later (so I can record the necessary information for the error), so I'll throw the exception after catching it